src/java.base/share/classes/sun/nio/ch/NioSocketImpl.java

```diff
@@ -248,6 +248,8 @@ private int tryRead(FileDescriptor fd, byte[] b, int off, int len)
         ByteBuffer dst = Util.getTemporaryDirectBuffer(len);
         assert dst.position() == 0;
         try {
+            // 'dst' is guaranteed not to be associated with a closeable session.
+            // Hence, there is no need for acquiring any session.
```

Comment:
This comment is will be confusing to anyone reading this code. Is this really needed?

Reply:
The reason for the comment is to make it clear why `DirectBuffer::address` can be used directly without guarding. This will also reduce the probability of unnecessary guarding being added in the future. However, if the consensus is that these comments just adds confusion, I am happy to remove them.

Reply:
I'd prefer to see this comment removed from all places that are obviously interacting with the direct buffer cache. These usages are try-finally to acquire and return the temporary direct buffer cache back to the cache. Talking about closable sessions here is definitely confusing.

Reply:
> Thanks for persisting with it, latest version looks okay to me and the naming issue can be sorted out after the JEP is integrated.


IMHO the renaming is not super important - the underlying abstraction managing the segment lifetime is still called MemorySession, even after https://git.openjdk.org/jdk/pull/10872. And, acquire/release are methods on MemorySession - so I think the current name might be more precise even after we integrate the latest API changes.